pagopa/io-ts-commons
# [#176013578] refactor pipeline

Implement CI/CD pipelines according to [our guidelines](https://pagopa.github.io/io-docs/io-handbook/cicd-setup/).

Added features

- create dedicated pipelines for deploy and code review
- automatic version bump during deploy
- automatic publication on npm registry

**To verify:** 
I tried to add the possibility to test the deploy pipeline into a dedicated branch (named _<xyz>-TEST-PIPELINE_)
In this case it should skip the process of publishing the artifact to npm, and it should create the changelog into the branch itself instead of master.
This feature needs to be tested and verified.
